include/albatross/src/evaluation/prediction_metrics.hpp

```diff
@@ -151,8 +151,23 @@ namespace detail {
 
 inline Eigen::MatrixXd principal_sqrt(const Eigen::MatrixXd &input) {
   const Eigen::SelfAdjointEigenSolver<Eigen::MatrixXd> eigs(input);
+  Eigen::VectorXd eigenvalues{eigs.eigenvalues()};
+  if (!(eigs.eigenvalues().array() >= 0.).all()) {
+    // In the unfortunate case of ill-conditioned arguments, which can easily
+    // happen when comparing two nearby distributions, we clamp
+    // numerically negative but morally OK eigenvalues to 0.
+    const double min_positive =
+        (eigenvalues.array() > 0)
+            .select(eigenvalues.array(),
+                    std::numeric_limits<double>::infinity())
+            .minCoeff();
+    eigenvalues =
+        (eigenvalues.array() < 0 && eigenvalues.array() > -10 * min_positive)
```

Comment:
Was this threshold derived through experimentation?

Reply:
No, I just didn't want to allow arbitrarily large negative eigenvalues to get clamped without complaint.